src/main/java/com/onarandombox/multiverseinventories/util/CommentedYamlConfiguration.java

```diff
@@ -213,63 +190,52 @@ public boolean save() {
     }
 
     /**
-     * Adds a comment just before the specified path.  The comment can be multiple lines.  An empty string will indicate a blank line.
+     * Adds a comment just before the specified path. The comment can be multiple lines. An empty string will indicate
+     * a blank line.
      *
      * @param path         Configuration path to add comment.
-     * @param commentLines Comments to add.  One String per line.
+     * @param commentLines Comments to add. One String per line.
      */
     public void addComment(String path, List<String> commentLines) {
-        StringBuilder commentstring = new StringBuilder();
-        String leadingSpaces = "";
+        StringBuilder commentString = new StringBuilder();
+        StringBuilder leadingSpaces = new StringBuilder();
         for (int n = 0; n < path.length(); n++) {
             if (path.charAt(n) == '.') {
-                leadingSpaces += "  ";
+                leadingSpaces.append("  ");
             }
         }
         for (String line : commentLines) {
             if (!line.isEmpty()) {
-                line = leadingSpaces + line;
-            } else {
-                line = " ";
+                line = leadingSpaces.toString() + line;
             }
-            if (commentstring.length() > 0) {
-                commentstring.append("\r\n");
+            if (commentString.length() > 0) {
+                commentString.append(System.getProperty("line.separator"));
```

Comment:
Isn't this the actual fix?

Reply:
No, I forget what the actual fix is though. I'll get back to you on this.

Reply:
The actual issue was the old line 73. The square brackets in the regex caused the resulting array to look something like this:
```
["Line 1...", "", "Line 2...", "", "Line 3", "", etc...
```
Removing the square brackets fixes the issue, then I added Lines 172 and 173 to keep gaps between config options with comments.